src/test/java/org/terasology/logic/inventory/InventoryManagerTest.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.logic.inventory;
+
+import com.google.common.collect.Sets;
+import org.junit.Assert;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.terasology.context.Context;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.moduletestingenvironment.ModuleTestingEnvironment;
+import org.terasology.world.block.BlockManager;
+import org.terasology.world.block.items.BlockItemFactory;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class InventoryManagerTest {
+
+    static final String URI_DIRT = "CoreAssets:Dirt";
+
+    private static ModuleTestingEnvironment context;
+
+    private Context hostContext;
+    private EntityManager entityManager;
+    private InventoryManager inventoryManager;
+    private BlockManager blockManager;
+    private EntityRef inventory;
+
+    @BeforeAll
+    public static void setup() throws Exception {
+        context = new ModuleTestingEnvironment() {
+            @Override
+            public Set<String> getDependencies() {
+                return Sets.newHashSet("CoreAssets", "Inventory");
+            }
+        };
+        context.setup();
+    }
+
+    @AfterAll
+    public static void tearDown() throws Exception {
+        context.tearDown();
+    }
+
+    @BeforeEach
+    public void beforeEach() {
+        hostContext = context.getHostContext();
+
+        entityManager = hostContext.get(EntityManager.class);
+        inventoryManager = hostContext.get(InventoryManager.class);
+        blockManager = hostContext.get(BlockManager.class);
+
+        inventory = entityManager.create(new InventoryComponent(3));
+    }
+
+    @ParameterizedTest
+    @ValueSource(bytes = {0, -1})
+    public void giveItem_zeroBlocks(int amount) {
+        EntityRef blockItem = getDirtBlockItem(entityManager, blockManager, amount);
+        inventoryManager.giveItem(inventory, EntityRef.NULL, blockItem);
+
+        final List<EntityRef> filledSlots = getFilledSlots(inventory);
+        Assert.assertEquals("No slots should be filled", 0, filledSlots.size());
```

Comment:
Use junit5's `Assertion` instead junit4's `Assert`. 
Most methods similar, except parameters order: message is last parameter.